etna/core/blocks/document.py

```diff
@@ -0,0 +1,53 @@
+from wagtail import blocks
+from wagtail.documents.blocks import DocumentChooserBlock
+
+
+class DocumentBlock(blocks.StructBlock):
+    """
+    A block for embedding a document file in a page.
+    """
+
+    link_text = blocks.CharBlock(
+        required=True,
+        help_text="The text to display for the link to the document",
+    )
+    file = DocumentChooserBlock(required=True)
+
+    def get_api_representation(self, value, context=None):
+        representation = super().get_api_representation(value, context)
+
+        file = value.get("file")
+
+        if file:
+            if file.file_size >= 1000000:
+                file_size = f"{((file.file_size / 1024) / 1024):.2f}MB"
+            else:
+                file_size = f"{(file.file_size / 1024):.2f}KB"
```

Comment:
Used 1024 (binary) representation for MB/KB instead of just 1000 - Windows file system (and I think all file systems) use 1024.

Reply:
I _think_ the current standard is 1,000 bytes = 1 kilobyte and 1,000 kilobytes = 1 megabyte which I personally hate because it has always been 1,024 for me. I think that 1,024 is for networking (when you talk about transfer speeds) and 1,000 is for storage.

There are a "new" set of silly sounding prefixes like "kibibyte", "mebibyte" and "gibibyte" which we are meant to use but honestly, I don't think a member of the public will know or care about these. (https://en.wikipedia.org/wiki/Binary_prefix#Definitions)

https://xkcd.com/394/ 😅

TL;DR - Can we use 1,000 base for now please?

Could we also please have two fields returned in the API; `file_size` which is simply the number of bytes and `pretty_file_size` which is the file size with suffix? I might need the "raw" number in the frontend.

We can use something like the below to produce a pretty size. (Please amend as necessary):

```python
def pretty_file_size(file_size):
    suffixes = ["B", "kB", "MB", "GB"]
    suffix_index = 0
    while file_size > 1000 and suffix_index < len(suffixes) - 1:
        file_size = file_size / 1000
        suffix_index = suffix_index + 1
    return f"{round(file_size, min(suffix_index, 2))}{suffixes[suffix_index]}"
```

Reply:
I was always taught kibibyte and kilobyte, but I think most people call a kibibyte (1024) kilobyte 😂 When I use 1000 base it can be off by quite a bit on my machine...

![image](https://github.com/nationalarchives/ds-wagtail/assets/62654785/e7a728f9-b323-4f1d-9901-5b613b3418db)
![image](https://github.com/nationalarchives/ds-wagtail/assets/62654785/09bf4363-49de-4951-8e15-41116dc1432e)

In this example 621.69MB is the same as the size of the file when it's on my machine, but if I do base 1000 it would be 651.87MB... I'm happy to go with either but I do think we should be keeping in line with the file system which is using kibi/mebi/gibibyte (1024), as it puts it out by quite a bit...

Reply:
![image](https://github.com/nationalarchives/ds-wagtail/assets/62654785/c47d9c74-0125-4f0c-923a-fcaca6cc7bbb)

![image](https://github.com/nationalarchives/ds-wagtail/assets/62654785/c7c7bbf6-2ee9-4065-adde-7c565b6fc654)

Reply:
I had to do a bit more reading up on where everything is with this flame war 😅

It looks like basically everyone except Microsoft use decimal (1,000) - this includes Mac OS, iOS, Ubuntu and Android. Also I just found out that weirdly the online Microsoft services like Office 365 use decimal as well.

https://www.simonrjones.net/2022/01/a-little-bit-about-filesize-units/#:~:text=Today%20Ubuntu%20Linux%2C%20Mac%20OS%2C%20iOS%20and%20Android%20use%20decimal%20for%20file%20storage%20sizes.%20Windows%2C%20as%20far%20as%20I%E2%80%99m%20aware%2C%20still%20uses%20binary%20units.%20However%2C%20to%20spice%20things%20up%20Microsoft%E2%80%99s%20cloud%20office%20service%20365%20uses%20decimal%20units%20when%20referring%20to%20cloud%20storage%20size!

I honestly think we should stick with decimal (1,000) and maybe once Microsoft stop turning Windows into an advertising platform and actually fix some stuff, they might move over to the correct units... 🤞🏼

Reply:
Good old Windows... 😅

I'll swap it over to 1000!